crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -2258,25 +2245,25 @@ buildSwitch :: (1 <= w)
             -> L.BlockLabel        -- ^ The label of the current basic block
             -> L.BlockLabel        -- ^ The label of the default basic block if no other branch applies
             -> [(Integer, L.BlockLabel)] -- ^ The switch labels
-            -> LLVMGenerator h s arch ret ()
+            -> LLVMGenerator h s arch ret (TermStmt s ret)
 buildSwitch _ _  curr_lab def [] =
    definePhiBlock curr_lab def
 buildSwitch w ex curr_lab def ((i,l):bs) = do
    let test = App $ BVEq w ex $ App $ BVLit w i
    test_a <- mkAtom test
-   endNow $ \_ -> do
-     t_id <- newLabel
-     f_id <- newLabel
-     endCurrentBlock $! Br test_a t_id f_id
+   t_id <- newLabelG
+   f_id <- newLabelG
+   resume_ (Br test_a t_id f_id) $ do
      defineBlock t_id (definePhiBlock curr_lab l)
-     defineBlock f_id (buildSwitch w ex curr_lab def bs)
+     return f_id
+   buildSwitch w ex curr_lab def bs
```

Comment:
Ugh, I'm not a fan of this pattern, nor the one above for conditional branches.

Would being able to define blocks directly in the Generator monad give a nicer API here?